sherpa/astro/fake.py

```diff
@@ -21,111 +21,158 @@
 
     Parameters
     ----------
     data : sherpa.astro.data.DataPHA
         The dataset (may be a background dataset).
-    model : sherpa.models.model.ArithmeticModel instance
-        The model that will be used for simulations.
-    is_source : bool
-        ``True`` means that the ``model`` does not contain response or
-        background components and that these need to be added based on the
-        ARF, RMF, and backgrounds set up for the data set. If ``False``, then
-        the ``model`` contains any components to describe the instrument
-        already.
-    pileup_model : None or `sherpa.astro.models.JDPileup` instance
-        Pileup Model for the source spectrum
-    add_bkgs : bool
-        If ``True`` backgrounds are added to the source counts.
-    bkg_srcs : dict
-        Keys in the dictionary need to be the background ids in the dataset
-        ``data``, and the values are the corresponsing source models. For all
-        background datasets that are listed in this dictionary, the
-        background counts will be simulated based on the model, appropriately
-        scaled (for area etc.) and added to the source. The same ``is_source``
-        setting as for the source model applies here, i.e. if the source model
-        already contains the ARF and the RMF, then the background models should
-        do so, too. This setting has no effect if ``add_bkgs=False``.
-
-        For all background ids not listed in this dictionary, the
-        counts will be drawn from the PHA data of the background data set.
-    id : str
-        String with id number if called from UI layer. This is only
-        used for certain error messages.
+    model : sherpa.models.model.ArithmeticModel
+        The model that will be used for simulations. It must contain
+        any background components, appropriately scaled, and include
+        the relevant response.
+    include_bkg_data : bool, optional
+        Should the counts in the background datasets be included when
+        calculating the predicted signal? As background datasets are
+        often noisy it is generally better to include a model of the
+        background in the model argument.
     method : callable or None
         The routine used to simulate the data. If None (the default)
-        then sherpa.utils.poisson_noise is used, otherwise the function
-        must accept a single ndarray, returning a ndarray of the same shape.
+        then sherpa.utils.poisson_noise is used, otherwise the
+        function must accept a single ndarray, returning a ndarray of
+        the same shape.
+
+    Notes
+    -----
+
+    The model must include the relevant response, and if a background
+    model is required then it must be included. The model is evaluated
+    to create the predicted counts. If the PHA dataset contains any
+    backgrounds and include_bkg_data is set then the counts are added,
+    after scaling, to the predicted counts. The predicted counts are
```

Comment:
the predicted counts ... ? From the model? i.e. if the model contains a background component and the backgrounds are set, do I get the background twice in the simulated data?

Reply:
I found it hard to distill the behaviour, so a good review of the docs is helpful! I keep on thinking that if we find the right way to do this then the description will become easier, but the various different possibilities for backgrounds makes this hard. 

I think I'd like to reduce the different possibilities but there may be some use for some of the current approaches, and almost-certainly someone using them, so I think that's a longer term goal than this PR.

Reply:
So maybe the approach is to leave it like this and address that in a follow-up PR? 
One approach might be to have different functions in the user facing API ("fake_pha_with_background_model", "fake_pha_no_bkg", "fake_pha_draw_from_bkg"), another might be to have a `background_method` argument that accepts more complex objects like `sherpa.fake.BackgroundFromModel`, `sherpa.data.NoBackground`, ... The user would then have to instantiate those objects separately, and thus they would have a smaller API that's hopefully easier to understand. Simulating background has become difficult here.

Reply:
The "predicted counts" on line 82 are meant to be the same as the "predicted counts" on line 80.

Reply:
I've reworked this section (so this first paragraph and then the next one). Whether it's better is hard to say.